### PR TITLE
Spacewalk cover overlay_connection and connector with unit tests

### PR DESCRIPTION
**Pull request by cr4pt0**

### CODE REVIEW

File: clients/stellar-relay/src/connection/connector/connector.rs [hunk 1]
```diff
@@ -104,7 +104,7 @@ impl Connector {
 			timeout_in_secs: cfg.timeout_in_secs,
 			retries: cfg.retries,
 			remote_called_us: cfg.remote_called_us,
-			receive_tx_messages: cfg.recv_scp_messages,
+			receive_tx_messages: cfg.recv_tx_msgs,
```
Comment (b-yap) [inline clients/stellar-relay/src/connection/connector/connector.rs:107]: nice catch!

File: clients/stellar-relay/src/connection/connector/connector.rs [hunk 2]
```diff
@@ -193,3 +193,218 @@ impl Connector {
...
+		assert_eq!(connector.remote().unwrap().sequence(), 0);
+
+		connector.increment_remote_sequence().unwrap();
+		connector.increment_remote_sequence().unwrap();
+		connector.increment_remote_sequence().unwrap();
+		assert_eq!(connector.remote().unwrap().sequence(), 3);
+	}
+
+	#[test]
+	fn connector_get_set_hmac_keys_works() {
```
Comment (ebma) [inline clients/stellar-relay/src/connection/connector/connector.rs:286]: I would change it to `connector_get_and_set_hmac_keys_works`

File: clients/stellar-relay/src/connection/connector/connector.rs [hunk 3]
```diff
@@ -193,3 +193,218 @@ impl Connector {
...
+		let cfg = ConnConfig::new("34.235.168.98", 11625, secret, 0, false, true, false);
+		// this is a channel to communicate with the connection/config (this needs renaming)
+		let (actions_sender, actions_receiver) = mpsc::channel::<ConnectorActions>(1024);
+		// this is a channel to communicate with the user/caller.
+		let (relay_message_sender, relay_message_receiver) =
+			mpsc::channel::<StellarRelayMessage>(1024);
+		let connector =
+			Connector::new(node_info.clone(), cfg.clone(), actions_sender, relay_message_sender);
+		(node_info, cfg, connector, actions_receiver, relay_message_receiver)
+	}
```
Comment (ebma) [inline clients/stellar-relay/src/connection/connector/connector.rs:409]: I would move these two functions to the top of the test module so they are known to the reader of the code before being used.

**Review by ebma — CHANGES_REQUESTED** [review]
There are some nice tests, good job. 👍 
Besides my other comments, can you also fix the warnings about unused warnings etc that are shown by CI?

**Comment by cr4pt0** [comment]
the problem is that ci shoe some wrong warning. 'create_connector' never used. But i use this function 6 times. What i can do with this one?

**Comment by ebma** [comment]
Yeah I also saw that and was wondering. Maybe it's fixed by moving it more to the top, as I suggested in my comment? If it does not fix it you can ignore it.

**Comment by b-yap** [comment]
@cr4pt0  maybe `#[cfg(test)]` on the function will work.

File: clients/stellar-relay/src/connection/overlay_connection.rs [hunk 4]
```diff
@@ -114,3 +114,127 @@ impl StellarOverlayConnection {
...
+			mpsc::channel::<StellarRelayMessage>(1024);
+
+		let mut overlay_connection = StellarOverlayConnection::new(
+			actions_sender.clone(),
+			relay_message_receiver,
+			cfg.retries,
+			node_info,
+			cfg,
+		);
+		let error_messae = "error message".to_owned();
```
Comment (ebma) [inline clients/stellar-relay/src/connection/overlay_connection.rs:188]: here is a typo

File: clients/stellar-relay/src/connection/connector/connector.rs [hunk 5]
```diff
@@ -193,3 +193,218 @@ impl Connector {
 		self.flow_controller.enable(local_overlay_version, remote_overlay_version)
 	}
 }
+
```
Comment (b-yap) [inline clients/stellar-relay/src/connection/connector/connector.rs:196]: ah. try the `#[cfg(test)]` here.

**Comment by ebma** [comment]
@b-yap ah true. This should be added above the declaration of the test module as it is missing right now.

**Review by b-yap — APPROVED** [review]
the check failure at line 87 of pallets/stellar-relay/src/libs.rs is out of scope. 

I'm approving this, based on the 2 files.

**Review by ebma — APPROVED** [review]

**Pull request merged**